### PR TITLE
Feature/#14 skeleton for events

### LINKED ISSUE #14: Define network communication data
**Describe the solution you'd like**
After a new game started, the host should create an initial game state. This game state includes money of players, player position and all other relevant information. When a user makes a move (rolles the dices, or cheats) he should send a message to the host. The host recalculate the gamestate and send it to the client's who repaint the gameboard.

### PR DESCRIPTION
**Pull request by valentingavran**
Closes #14

### CODE REVIEW

**Comment by valentingavran** [comment]
The code smells only exist because of the comments, which indicate how to proceed with the next step. This next step will be to build the requests and responses using the serialization methods implemented by Hannes, which converts the Java objects to JSON objects.

After that, only the type of requests or responses has to be determined and the respective method has to be called.

File: app/src/main/java/com/ss20/se2/monopoly/Utils.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
+package com.ss20.se2.monopoly;
+
+import com.google.gson.JsonArray;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class Utils{
```
Comment (dermutzh) [inline app/src/main/java/com/ss20/se2/monopoly/Utils.java:8]: Remove this class. I've implemented it in my PR and will be further specified/implemented in the next Sprint. This only leads to merge conflicts and it isn't currently used at all.

File: app/src/main/java/com/ss20/se2/monopoly/network/client/ResponseHandler.java [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+	 * Takes the response type and possible serialized Java objects from the response and
+	 * updates the database. After that this method notifies all listeners who have subscribed to changes.
+	 * Only methods of the listeners that are relevant are called. The shortly before changed data in
+	 * the database is  taken from there and added to the relevant methods as parameters.
+	 *
+	 * @param response
+	 */
+	private void processResponse(JsonObject response){
+		Log.d(NetworkUtilities.TAG, "Processing of response started");
+		//		ResponseType type = ResponseType.toEnum(response.get(...).getAsString());
```
Comment (dermutzh) [inline app/src/main/java/com/ss20/se2/monopoly/network/client/ResponseHandler.java:83]: Just as Info: Normally it is recommended to remove code which isn't needed.

File: app/build.gradle [hunk 3]
```diff
@@ -57,10 +57,14 @@ dependencies {
...
     implementation 'androidx.constraintlayout:constraintlayout:1.1.3'
 
+    // Library that converts Java Objects into their JSON representation and back
+    implementation 'com.google.code.gson:gson:2.8.6'
+
     // Testing
     testImplementation 'junit:junit:4.12'
     androidTestImplementation 'androidx.test.ext:junit:1.1.1'
     testImplementation 'org.mockito:mockito-inline:2.27.0'
+    testImplementation 'org.json:json:20140107'
```
Comment (dermutzh) [inline app/build.gradle:67]: remove this dependency, it is never used.

File: app/src/main/java/com/ss20/se2/monopoly/network/client/ResponseHandler.java (hunk 2 above)
Comment (valentingavran) [reply app/src/main/java/com/ss20/se2/monopoly/network/client/ResponseHandler.java:83]: Thanks for the tip, did not know what is common in the industry. Should I also remove the JavaDoc documentation of the method?

**Comment by bferijan** [comment]
Everything looks fine. A lot of events habe already been "created" but there might be some more. The idea with the interface for server and client was good.

**Pull request merged**